chain/client/src/client.rs

```diff
@@ -2004,4 +2012,63 @@ impl Client {
             next_blocks_log.join("\n")
         ))
     }
+
+    pub fn detailed_upcoming_blocks_info_as_web(&self) -> ChunkInfoView {
+        let height_status_map = match self.detailed_upcoming_blocks_info() {
+            Ok(m) => m,
+            Err(_) => HashMap::new(),
+        };
+
+        let next_blocks_log = height_status_map
+            .keys()
+            .sorted()
+            .map(|height| {
+                let val = height_status_map.get(height).unwrap();
```

Comment:
As we need to fetch a value here anyway, we can use `.iter()` rather than `.keys()` to avoid `.unwrap()` (which would make *other* bugs stand out more) and to reduce the number of map lookups we need (improving perf a bit).